spyder/plugins/editor/tests/test_introspection.py

```diff
@@ -99,6 +99,20 @@ def test_space_completion(setup_editor):
     assert code_editor.toPlainText() == 'from numpy import\n'
 
 
+@pytest.mark.slow
+def test_hide_widget_completion(setup_editor):
+    """Validate hiding completion widget after a delimeter or operator."""
+    editor, qtbot = setup_editor
+    code_editor = editor.get_focus_widget()
+    completion = code_editor.completion_widget
+
+    # Set cursor to start
+    code_editor.go_to_line(1)
+
+    qtbot.keyClicks(code_editor, 'from numpy [')
```

Comment:
@steff456, this is not testing things correctly. First we need to make the completion widget appear, then write `[` and finally verify that the widget is not visible.

Reply:
I'm going to check it!